R/nm-join.R

```diff
@@ -0,0 +1,179 @@
+#' Return a single data frame with model output and input data
+#'
+#' For NONMEM models, when a unique row identifier (e.g an integer numbering the
+#' rows) is included in the input data set (i.e. the file in `$DATA`) and
+#' carried into each table output, `nm_join()` can read in all output table
+#' files and join back to the input data set. By default, the input data is
+#' joined to the table files so that the number of rows in the result will match
+#' the number of rows in the table files (i.e. the number of rows _not_ bypassed via
+#' `$IGNORE`). Use the `.superset` argument to join table outputs to the
+#' (complete) input data set. This function will print the number of rows and
+#' columns when each file is loaded, as well as some information about the
+#' joins. This **printing can be suppressed** by setting `options(bbr.verbose =
+#' FALSE)`.
+#'
+#' @inheritParams nm_tables
+#' @param .join_col Character column name to use to join table files. Defaults to
+#'   `NUM`. See Details.
+#' @param .superset If `FALSE`, the default, the data will be joined to the
+#'   NONMEM output and if `TRUE`, the NONMEM output will be joined to the data;
+#'   that is, if you use `.superset`, you will get the same number of rows as
+#'   you have in the input data and NONMEM output columns like `PRED` and
+#'   `CWRES` will be filled with `NA`.
+#' @param .bbi_args Named list passed to `model_summary(.bbi_args)`. See
+#'   [print_bbi_args()] for valid options. Defaults to `list(no_grd_file = TRUE,
+#'   no_shk_file = TRUE)` because [model_summary()] is only called internally to
+#'   extract the number of records and individuals, so those files are
+#'   irrelevant.
+#'
+#' @details
+#'
+#' **Join column**
+#'
+#' The `.join_col` is the name of a single column that should appear in both the
+#' input data set and any tables you want to join. We recommend you make this
+#' column a simple integer numbering the rows in the input data set (for example
+#' `NUM`). When this column is carried into the output table files, there will
+#' be unambiguous matching from the table file back to the input data set.
+#'
+#' The one exception to this are `$FIRSTONLY` tables. If a table file has the
+#' same number of rows as the there are individuals in the input data set, it
```

Comment:
"same number of rows as the there are individuals in the input data set (accounting for any filtering of data in the NONMEM control stream)."